packages/main/src/RPA/Robocorp/utils.py

```diff
@@ -113,9 +113,15 @@ def handle_error(response: requests.Response):
 
     # pylint: disable=no-self-argument
     def _needs_retry(exc: BaseException) -> bool:
-        # Don't retry on server (500/internal/unexpected) and auth errors (401/403).
-        no_retry_codes = [400, 401, 403, 409, 500]
-        no_retry_messages = ["ERR_UNEXPECTED"]
+        # Don't retry on some specific error codes or messages.
+
+        # https://www.restapitutorial.com/httpstatuscodes.html
+        # 400 - payload is bad and needs to be changed
+        # 401 - missing auth bearer token
+        # 403 - auth is in place, but not allowed (insufficient privileges)
+        # 409 - payload not good for the affected resource
+        no_retry_codes = [400, 401, 403, 409]
+        no_retry_messages = []
```

Comment:
Looks good to me 👍

Reply:
Talked about this with @bendersej  and it makes more sense to just retry by default since we may encounter issues related to networking itself like timeouts, proxy issues or broken pipes and we can easily miss out cases like these.
So is better to cover explicitly what we know for sure we shouldn't try and retry everything else without overthinking it :).

Reply:
Sure, makes sense